src/plugins/httpUpload/HttpUploadPlugin.ts

```diff
@@ -246,14 +290,86 @@ export default class HttpUploadPlugin extends AbstractPlugin {
 
             if (/^data:image\/(jpeg|jpg|gif|png|svg);base64,[/+=a-z0-9]+$/i.test(thumbnailData) && /^https?:\/\//.test(url)) {
                let attachment = new Attachment(name, mimeType, url);
-               attachment.setThumbnailData(thumbnailData);
+
+               let domain = this.extractDomainFromUrl(url);
+               if (this.isTrustedDomain(domain))
+               {
+                  attachment.setThumbnailData(thumbnailData);
+               }
+               else
+               {
+                  attachment.setGenericThumbnail();
+               }
+
                attachment.setData(url);
 
                message.setAttachment(attachment);
-               message.setPlaintextMessage(bodyElement.text());
+
+               if (bodyElement.text()!=='')
+               {
+                   message.setPlaintextMessage(bodyElement.text());
+               }
+               else
+               {
+                   //Check if in normal body tag is text after URL to show
+                   if (body.text().split('\n').length>0)
+                   {
+                       let bodyTextArr = body.text().split('\n');
+                       bodyTextArr[0]='';
+                       let bodyText = bodyTextArr.map(item => item.trim()).join('\n').trim();
+                       message.setPlaintextMessage(bodyText);
+                   }
+               }
             }
          }
       }
+      else
+          if (body.length===1 && !message.isEncrypted())
+          {
+              let link = body.text();
+              let bodyText = link;
+              //Check if in normal body tag is text after URL to show
+              if (bodyText.split('\n').length>0)
```

Comment:
Why are we requiring a newline to display the image?

Reply:
i process all links now...